ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -760,27 +765,98 @@ public void shouldReturn5xxOnStatementSystemError() {
   }
 
   @Test
-  public void shouldFailOnSetProperty() {
+  public void shouldSetProperty() {
+    // Given:
+    final String csas = "CREATE STREAM " + streamName + " AS SELECT * FROM test_stream;";
+
     // When:
-    final KsqlErrorMessage result = makeFailingRequest(
-        "SET 'auto.offset.reset' = 'earliest';", Code.BAD_REQUEST);
+    final List<CommandStatusEntity> results = makeMultipleRequest(
+        "SET '" + KsqlConfig.SINK_NUMBER_OF_REPLICAS_PROPERTY + "' = '2';\n"
+            + csas,
+        CommandStatusEntity.class);
 
     // Then:
-    assertThat(result.getMessage(), is(
-        "SET and UNSET commands are not supported on the REST API. "
-            + "Pass properties via the 'streamsProperties' field"));
+    verify(commandStore).enqueueCommand(eq(csas), any(), any(),
+        eq(ImmutableMap.of(KsqlConfig.SINK_NUMBER_OF_REPLICAS_PROPERTY, "2")));
+
+    assertThat(results, hasSize(1));
+    assertThat(results.get(0).getStatementText(), is(csas));
   }
 
   @Test
-  public void shouldFailOnUnsetProperty() {
+  public void shouldFailSetPropertyOnInvalidPropertyName() {
     // When:
-    final KsqlErrorMessage result = makeFailingRequest(
-        "UNSET 'auto.offset.reset';", Code.BAD_REQUEST);
+    final KsqlErrorMessage response = makeFailingRequest(
+        "SET 'ksql.unknown.property' = '1';",
+        Code.BAD_REQUEST);
 
     // Then:
-    assertThat(result.getMessage(), is(
-        "SET and UNSET commands are not supported on the REST API. "
-            + "Pass properties via the 'streamsProperties' field"));
+    assertThat(response, instanceOf(KsqlStatementErrorMessage.class));
+    assertThat(response.getErrorCode(), is(Errors.ERROR_CODE_BAD_STATEMENT));
+    assertThat(response.getMessage(), containsString("Unknown property"));
+  }
+
+  @Test
+  public void shouldFailSetPropertyOnInvalidPropertyValue() {
+    // When:
+    final KsqlErrorMessage response = makeFailingRequest(
+        "SET '" + KsqlConfig.SINK_NUMBER_OF_REPLICAS_PROPERTY + "' = 'invalid value';",
+        Code.BAD_REQUEST);
+
+    // Then:
+    assertThat(response, instanceOf(KsqlStatementErrorMessage.class));
+    assertThat(response.getErrorCode(), is(Errors.ERROR_CODE_BAD_STATEMENT));
+    assertThat(response.getMessage(),
+        containsString("Invalid value invalid value for configuration ksql.sink.replicas: "
+            + "Not a number of type SHORT"));
+  }
+
+  @Test
+  public void shouldUnsetProperty() {
```

Comment:
What's the expected behavior for unsetting a property that wasn't provided/set? From the current implementation it looks like it's a no-op. Perhaps add a test to confirm no error is thrown?